libs/ui-lib/lib/ocm/components/clusterWizard/CredentialsDownload.tsx

```diff
@@ -0,0 +1,137 @@
+import { Alert, AlertVariant, Checkbox, Grid } from '@patternfly/react-core';
+import { downloadFile, getErrorMessage } from '../../../common/utils';
+import { ExternalLinkAltIcon } from '@patternfly/react-icons/dist/js/icons/external-link-alt-icon';
+
+const CredentialsDownload: React.FC<{ cluster: Cluster }> = ({ cluster }) => {
+  const [isChecked, setIsChecked] = React.useState<boolean>(false);
+  const clusterWizardContext = useClusterWizardContext();
+  const [credentials, setCredentials] = React.useState<Credentials>();
+  const [credentialsError, setCredentialsError] = React.useState('');
+  const [isDownloading, setIsDownloading] = React.useState(false);
+  const { addAlert } = useAlerts();
+  const { t } = useTranslation();
+
+  const fetchCredentials = React.useCallback(() => {
+    const fetch = async () => {
+      setCredentialsError('');
+      if (!cluster.id) {
+        return;
+      }
+      try {
+        const response = await ClustersAPI.getCredentials(cluster.id);
+        const data = { username: response.data.username, password: response.data.password };
+        setCredentials(data);
+      } catch (err) {
+        setCredentialsError(`Failed to fetch credentials, ${getErrorMessage(err)}`);
+      }
+    };
+    void fetch();
+  }, [cluster]);
+
+  React.useEffect(() => {
+    fetchCredentials();
+  }, [fetchCredentials]);
+
+  const downloadSingleFile = async (fileName: string) => {
+    try {
+      const response = await ClustersAPI.downloadClusterCredentials(cluster.id, fileName);
+      downloadFile('', response.data, fileName);
+      return true;
+    } catch (e) {
+      handleApiError(e, (e) => {
+        addAlert({
+          title: t(`Could not download ${fileName} file`),
+          message: getApiErrorMessage(e),
+        });
+      });
+      return false;
+    }
+  };
+
+  const handleDownloadClick = async () => {
+    setIsDownloading(true);
+
+    const configSuccess = await downloadSingleFile('kubeconfig');
+    const passwordSuccess = await downloadSingleFile('kubeadmin-password');
+
+    const hasError = !configSuccess || !passwordSuccess;
+
+    setTimeout(() => {
+      setIsDownloading(false);
+
+      if (!hasError) {
+        clusterWizardContext.moveNext();
+      }
+    }, 500);
+  };
+
+  const footer = (
+    <ClusterWizardFooter
+      cluster={cluster}
+      onNext={() => void handleDownloadClick()}
+      onBack={() => clusterWizardContext.moveBack()}
+      isNextDisabled={!isChecked || isDownloading}
+      nextButtonText="Download credentials"
+      isNextButtonLoading={isDownloading}
+    />
+  );
+
+  return (
+    <ClusterWizardStep navigation={<ClusterWizardNavigation cluster={cluster} />} footer={footer}>
+      <WithErrorBoundary title={t('ai:Failed to load Credentials Download step')}>
+        <ClusterWizardStepHeader>Download credentials</ClusterWizardStepHeader>
+        <Alert
+          style={{ marginTop: '20px', marginBottom: '20px' }}
```

Comment:
please don't use custom styling when possible.

Instead use the `Layouts` from PF
for example `<Stack hasGutter>` will do just fine here